File: HW01/22001641_karatas_emre.dart
void main() 
{
  /* In void main()...
   * declare a string list*/
  List<String> emptyStringList = [];
  
  //Declaring and initializing famous CS people in the history in a list.
  List<String> famousCSPeople = ['Tim Berners-Lee', 'Alan Turing', 'Barbara Liskov'];
  
  // checking whether the given list is empty or not.
  bool famousCSPeopleChecking = famousCSPeople.isEmpty;
  
  // also check emptyStringList declared above
   bool emptyStringListChecking = emptyStringList.isEmpty;
  
  // printing result of bool value computed above.
  famousCSPeopleChecking ? print("famousCSPeople empty? :true"): print("famousCSPeople empty? : false");
  
  // printing result of bool value computed above.
  emptyStringListChecking ? print("emptyStringList empty? :true"): print("emptyStringList empty? :false");
  
  
  // adding new element to list. Lars Bak is the inventor of Dart language.
  famousCSPeople.add('Lars Bak');
  
  /*check whether list contains particular element or not 
   * Ada Lovalace is the inventor of ADA Language
   * and it is not in the famousCSPeople list.*/
  bool containsAdaLovalace = famousCSPeople.contains('Ada Lovalace');
  
  // printing bool result by ternary operator
  containsAdaLovalace ? print("famousCSPeople includes Ada Lovalace? :true"): print("famousCSPeople includes Ada Lovalace? :false");
  
  /*remove a particular element from the list 
  Barbara Liskov is an element of famousCSPeople list*/
  famousCSPeople.remove('Barbara Liskov');
  
  // get head and tail of the famousCSPeople list
  var listHead = famousCSPeople.first;
  var listTail = famousCSPeople.sublist(1);
  
  // prints the first element of the list
  print("List head: $listHead"); 
  // prints all elements of the list except for the first one
  print("List tail: $listTail"); 

  print("List elements:"); 
  // prints all elements in the famousCSPeople list
  for (var elements in famousCSPeople) 
  {
    print(elements);
  }

}

File: HW01/22001641_karatas_emre.go
package main

import (
	"container/list"
	"fmt"
)

func main() {

    //creating empty list
	emptyList := list.New()
    
    //creating famousCSPeople list.
	famousCSPeople := list.New()
    
    //populating famousCSPeople list by elements
	famousCSPeople.PushBack("Tim Berners-Lee")
	famousCSPeople.PushBack("Alan Turing")
	famousCSPeople.PushBack("Barbara Liskov")
    
    //getting the length of the famousCSPeople list to determine whether it is empty or not
	famousCSPeopleChecking := famousCSPeople.Len() == 0
    
    //getting the length of the emptyList list to determine whether it is empty or not
	emptyListChecking := emptyList.Len() == 0
    
     //printing results for emptyList
	if emptyListChecking {
		fmt.Println("The emptylist is empty.")
	} else {
		fmt.Println("The emptylist is not empty.")
	}

    //printing results for famousCSPeople
	if famousCSPeopleChecking {
		fmt.Println("The famousCSPeople list is empty.")
	} else {
		fmt.Println("The famousCSPeople list is not empty.")
	}

    // adding new element to famousCSPeople list
	famousCSPeople.PushBack("Lars Bak")

    //checking whether the particular element exists in the famousCSPeople list or not.
	element := "Ada Lovelace"
	containsAdaLovalace := false

	for e := famousCSPeople.Front(); e != nil; e = e.Next() {
		if e.Value == element {
			containsAdaLovalace = true
			break
		}
	}

    // printing results for element existing result for the famousCSPeople list.
	if containsAdaLovalace {
		fmt.Println("famousCSPeople contains Ada Lovalace")
	} else {
		fmt.Println("famousCSPeople does not contain Ada Lovalace")
	}

    //removing particular element from the famousCSPeople list
	elementToRemove := "Barbara Liskov"

	for e := famousCSPeople.Front(); e != nil; e = e.Next() {
		if e.Value == elementToRemove {
			famousCSPeople.Remove(e)
			break
		}
	}

   //getting head of the famousCSPeople list and printing it
   if famousCSPeople.Len() > 0 {
    head := famousCSPeople.Front().Value
    fmt.Println("Head is:", head)
    }

    //getting tail slice of the famousCSPeople list and printing it
    // slicing is used to get tail list of the famousCSPeople list.
    if famousCSPeople.Len() > 1 {
    tail := make([]interface{}, 0)
        for e := famousCSPeople.Front().Next(); e != nil; e = e.Next() {
            tail = append(tail, e.Value)
        }
    fmt.Println("Tail is:", tail)
    }

    
    //printing all elements in the famousCSPeople list
	for e := famousCSPeople.Front(); e != nil; e = e.Next() {
		fmt.Println(e.Value)
	}

}
